### PR TITLE
httpd: Add patch head branches to patch output

### PR DESCRIPTION
**Pull request by sebastinez**
This PR adds the `RefStrings` of the branches that have the patch head.

### CODE REVIEW

File: radicle-httpd/src/api/json.rs [hunk 1]
```diff
@@ -134,6 +137,24 @@ fn name_in_path(path: &str) -> &str {
...
+    head: &Oid,
+) -> Result<Vec<RefString>, refs::Error> {
+    let remote = repo.remote(id)?;
+    let branches = remote
+        .refs
+        .iter()
+        .filter_map(|(name, o)| match o == head {
+            true => Some(name.to_owned()),
+            false => None,
+        })
```
Comment (cloudhead) [inline radicle-httpd/src/api/json.rs:152]: Maybe you can use an `if` instead of the `match`?

File: radicle-httpd/src/api/json.rs [hunk 2]
```diff
@@ -117,6 +119,7 @@ pub(crate) fn patch(id: PatchId, patch: Patch) -> Value {
                 "description": rev.description(),
                 "base": rev.base,
                 "oid": rev.oid,
+                "branches": get_branches(repo, patch.author().id(), &rev.oid).unwrap_or(vec![]),
```
Comment (cloudhead) [inline radicle-httpd/src/api/json.rs:122]: So one important detail is that these may not all be branches. I think tags could be returned as well here, or refs that aren't branches.

I'd suggest calling this key `refs` for now, to be correct. Later, we could filter it to only include branches, but it's a bit trickier.

Comment (cloudhead) [reply radicle-httpd/src/api/json.rs:122]: Also would change the function to `get_refs`

File: radicle-httpd/src/api/json.rs (hunk 1 above)
Comment (sebastinez) [reply radicle-httpd/src/api/json.rs:155]: fixed in e438753

Comment (cloudhead) [reply radicle-httpd/src/api/json.rs:155]: Hehe it's a pretty common pattern :shrug: 

There's also https://doc.rust-lang.org/std/primitive.bool.html#method.then_some

**Review by cloudhead — APPROVED** [review]

File: radicle-httpd/src/api/json.rs (hunk 1 above)
Comment (sebastinez) [reply radicle-httpd/src/api/json.rs:155]: Oh nice! Will jot that down

**Pull request merged**